src/builder_state.rs

```diff
@@ -679,57 +663,32 @@ impl<TYPES: NodeType> BuilderState<TYPES> {
     ) -> Self {
         let txns_in_queue: HashSet<_> = tx_queue.iter().map(|tx| tx.commit).collect();
         BuilderState {
-            included_txns: HashSet::new(),
-            included_txns_old: HashSet::new(),
-            included_txns_expiring: HashSet::new(),
+            num_nodes,
             txns_in_queue,
             built_from_proposed_block,
-            decide_receiver,
-            da_proposal_receiver,
-            qc_receiver,
             req_receiver,
-            da_proposal_payload_commit_to_da_proposal: HashMap::new(),
-            quorum_proposal_payload_commit_to_quorum_proposal: HashMap::new(),
-            tx_receiver,
             tx_queue,
             global_state,
-            builder_commitments: HashSet::new(),
-            total_nodes: num_nodes,
             maximize_txn_capture_timeout,
             base_fee,
             instance_state,
-            txn_garbage_collect_duration,
-            next_txn_garbage_collect_time: Instant::now() + txn_garbage_collect_duration,
             validated_state,
+            included_txns: RotatingSet::new(txn_garbage_collect_duration),
+            da_proposal_payload_commit_to_da_proposal: HashMap::new(),
+            quorum_proposal_payload_commit_to_quorum_proposal: HashMap::new(),
+            builder_commitments: HashSet::new(),
+            decide_receiver: receivers.decide.activate_cloned(),
+            da_proposal_receiver: receivers.da_proposal.activate_cloned(),
+            qc_receiver: receivers.quorum_proposal.activate_cloned(),
+            tx_receiver: receivers.transactions.activate_cloned(),
```

Comment:
Nit: Can just pass in `receivers`.

Reply:
Not really, `receivers` contains `InactiveReceiver`s instead of just `Receiver`s, so we need to activate them.
My idea was that with `BroadcastReceivers` containing `InactiveReceiver`s the application can now keep `BroadcastReceivers` around if they are needed at some point, for example for testing or something